src/Nethermind/Nethermind.JsonRpc/Modules/Admin/AdminRpcModule.cs

```diff
@@ -156,8 +156,20 @@ public async Task<ResultWrapper<bool>> admin_removeTrustedPeer(string enode)
     }
 
     public ResultWrapper<PeerInfo[]> admin_peers(bool includeDetails = false)
-        => ResultWrapper<PeerInfo[]>.Success(
-            _peerPool.ActivePeers.Select(p => new PeerInfo(p.Value, includeDetails)).ToArray());
+    {
+        var validatedPeers = _peerPool.ActivePeers
+            .Where(p => IsValidatedPeer(p.Value))
+            .Select(p => new PeerInfo(p.Value, includeDetails))
+            .ToArray();
+
+        return ResultWrapper<PeerInfo[]>.Success(validatedPeers);
+    }
+
+    private static bool IsValidatedPeer(Peer peer)
+    {
+        return peer.InSession?.IsNetworkIdMatched == true ||
+               peer.OutSession?.IsNetworkIdMatched == true;
```

Comment:
@asdacap why are even those peers there? Is it because for short period of time we don't know about their network id? Should PeerPool filter them internally?

Reply:
Yes, we don't know them temporarily.